### PR TITLE
[DOCS-7528] adding support for ASM Single step

### PR DESCRIPTION
**Pull request by aliciascott**
### What does this PR do? What is the motivation?
[DOCS-7528](https://datadoghq.atlassian.net/browse/DOCS-7528)

### Merge instructions

- [ ] Please merge after reviewing

### Additional notes

[DOCS-7528]: https://datadoghq.atlassian.net/browse/DOCS-7528?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by robertpi — APPROVED** [review]
Preview looks good to me.

File: content/en/security/application_security/enabling/tracing_libraries/_index.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+title: Enabling ASM using Datadog Tracing Libraries
+kind: documentation
+disable_toc: true
+---
+
+## Overview
+
+Instrumentation allows you to quickly get security insights from a wide range of standard operations and popular frameworks with minimal manual intervention. You can automatically instrument your ASM libraries when you install the Datadog Agent with single step instrumentation, or when you [manually add Datadog tracing libraries][1] to your code.
```
Comment (brett0000FF) [inline content/en/security/application_security/enabling/tracing_libraries/_index.md:9]: Same feedback as Enabling ASM using Single Step Instrumentation.

File: content/en/security/application_security/enabling/tracing_libraries/code_security/dotnet.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+To leverage code security vulnerability detection capabilities for your service:
+
+1. [Update your Datadog Agent][3] to at least version 7.41.1.
+2. Update your tracing library to at least the minimum version needed to turn on code security vulnerability detection. For details, see [ASM capabilities support][4].
+3. Add the `DD_IAST_ENABLED=true` environment variable to your application configuration. For example, on Windows self-hosted, run the following PowerShell snippet as part of your application start-up script:
+
+```sh
+$target=[System.EnvironmentVariableTarget]::Process
+[System.Environment]::SetEnvironmentVariable("DD_IAST_ENABLED","true",$target)
+```
```
Comment (brett0000FF) [inline content/en/security/application_security/enabling/tracing_libraries/code_security/dotnet.md:45]: I recommend indenting this under step 3.

File: content/en/security/application_security/enabling/tracing_libraries/code_security/_index.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+---
+title: Enabling ASM Code Security using Datadog Tracing Libraries
+kind: documentation
+type: multi-code-lang
+further_reading:
```
Comment (brett0000FF) [inline content/en/security/application_security/enabling/tracing_libraries/code_security/_index.md:5]: Heads up: The index pages you have are missing the further reading partial at the bottom, so they aren't displaying.

File: content/en/security/application_security/enabling/tracing_libraries/code_security/_index.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+- link: "/security/application_security/"
+  tag: "Documentation"
+  text: "Protect against Threats with Datadog Application Security Management"
+- link: "/security/application_security/enabling/compatibility/"
+  tag: "Documentation"
+  text: "Programming Language and Framework Compatibility"
+- link: "/security/application_security/add-user-info/"
+  tag: "Documentation"
+  text: "Tracking user activity"
+- link: "/security/default_rules/#cat-application-security"
```
Comment (brett0000FF) [inline content/en/security/application_security/enabling/tracing_libraries/code_security/_index.md:15]:
```suggestion
- link: "/security/default_rules/?category=cat-application-security"
```
^ This is in a few of your further reading sections, but I think the URL is incorrect.

File: content/en/security/application_security/enabling/tracing_libraries/code_security/nodejs.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
+---
+title: Enabling ASM for NodeJs
```
Comment (brett0000FF) [inline content/en/security/application_security/enabling/tracing_libraries/code_security/nodejs.md:2]:
```suggestion
title: Enabling ASM for Node.js
```

File: content/en/security/application_security/enabling/tracing_libraries/threat_detection/_index.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  text: "Troubleshooting Application Security Management"
+- link: "/security/application_security/how-appsec-works/"
+  tag: "Documentation"
+  text: "How Application Security Management Works in Datadog"
+- link: "https://www.datadoghq.com/blog/secure-serverless-applications-with-datadog-asm/"
+  tag: "Blog"
+  text: "Secure serverless applications with Datadog ASM"
+---
+
+Select your application language for details on how to implement these steps for your language and infrastructure types.
```
Comment (brett0000FF) [inline content/en/security/application_security/enabling/tracing_libraries/threat_detection/_index.md:29]:
```suggestion
Select your application language for details on how to enable ASM Thread Detection for your language and infrastructure types.
```

**Review by brett0000FF — APPROVED** [review]
Looks great! Left a few comments and suggestions. Please let me know if you have any questions or need any clarification. 🚀

File: content/en/security/application_security/enabling/_index.md [hunk 7]
```diff
@@ -39,15 +39,24 @@ Enable your application to [detect and protect against threats][1] targeting you
 
 {{% appsec-getstarted-with-rc %}}
 
-Select your application language for details on how to do these steps for your language and infrastructure types.
+## ASM enablement types
 
-{{< partial name="security-platform/appsec-languages.html" >}}
+There are two main approaches to enable ASM on your tracing libraries: single-step or with Datadog tracing libraries.
+
+### Single-step enablement
```
[the remark this answers is not in the document]
Comment (aliciascott) [reply content/en/security/application_security/enabling/_index.md:46]: Ah good point, I might have it different in a few places, the feature itself is combined "Single Step Instrumentation" for both APM + ASM, so I'll update it to instrumentation where needed.

File: content/en/security/application_security/enabling/single_step/_index.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+title: Enabling ASM using Single Step Instrumentation 
+kind: documentation
+disable_toc: true
+---
+
+<div class="alert alert-info">Enabling ASM using single step instrumentation is in beta.</div>
+
+## Overview
+
+Instrumentation allows you to quickly get security insights from a wide range of standard operations and popular frameworks with minimal manual intervention. You can automatically instrument your ASM libraries when you install the Datadog Agent with single step instrumentation, or when you [manually add Datadog tracing libraries][1] to your code.
```
[the remark this answers is not in the document]
Comment (aliciascott) [reply content/en/security/application_security/enabling/single_step/_index.md:11]: Yes that's correct, good catch !

File: content/en/security/application_security/enabling/tracing_libraries/code_security/_index.md (hunk 3 above)
Comment (aliciascott) [reply content/en/security/application_security/enabling/tracing_libraries/code_security/_index.md:5]: thank you!

**Comment by aliciascott** [comment]
Thanks @brett0000FF , since I made so many additional updates , do you mind doing a 2nd pass on my last commit when it finishes ?

**Review by brett0000FF — APPROVED** [review]
Looks good!

File: content/en/security/application_security/enabling/single_step/code_security.md [hunk 9]
```diff
@@ -0,0 +1,420 @@
...
+   ```
+   datadog:
+    apiKeyExistingSecret: datadog-secret
+    site: <DATADOG_SITE>
+    apm:
+      instrumentation:
+         enabled: true
+    env:
+      - name: DD_IAST_ENABLED=true 
+        value: "true"
```
Comment (eliottness) [inline content/en/security/application_security/enabling/single_step/code_security.md:209]: I don't think this part is going to work as is.

Looking at the docs tells me that `datadog.env` field adds this environment variable on all *agent* pods but this environment variable should be sent to all *traced* pods for this to work.

I think what you are looking for is:
```suggestion
    asm:
      iast:
        enabled: true
```

Cf. The [PR](https://github.com/DataDog/helm-charts/pull/1337/files) I made in the helm charts that should be merged once the new version of the cluster agent is released.

File: content/en/security/application_security/enabling/single_step/threat_detection.md [hunk 10]
```diff
@@ -0,0 +1,420 @@
...
+   ```
+   datadog:
+    apiKeyExistingSecret: datadog-secret
+    site: <DATADOG_SITE>
+    apm:
+      instrumentation:
+         enabled: true
+    env:
+      - name: DD_APPSEC_ENABLED
+        value: "true"
```
Comment (eliottness) [inline content/en/security/application_security/enabling/single_step/threat_detection.md:209]: ☝️ Same comment as above but with an other suggestion:
```suggestion
    asm:
      threats:
        enabled: true
```

File: content/en/security/application_security/enabling/single_step/threat_detection.md [hunk 11]
```diff
@@ -0,0 +1,420 @@
...
+     apiKeyExistingSecret: datadog-secret
+     site: <DATADOG_SITE>
+     tags:
+         - env:staging
+     apm:
+       instrumentation:
+         enabled: true
+     env:
+      - name: DD_APPSEC_ENABLED
+        value: "true"
```
Comment (eliottness) [inline content/en/security/application_security/enabling/single_step/threat_detection.md:314]:
```suggestion
     asm:
       threats:
         enabled: true
```

File: content/en/security/application_security/enabling/single_step/threat_detection.md [hunk 12]
```diff
@@ -0,0 +1,420 @@
...
+       metadata:
+         labels:
+           admission.datadoghq.com/enabled: "false"
+   ```
+2. Apply the configuration:
+   ```bash
+   kubectl apply -f /path/to/your/deployment.yaml
+   ```
+3. Restart the services you want to remove instrumentation for.
+
```
Comment (eliottness) [inline content/en/security/application_security/enabling/single_step/threat_detection.md:371]: I believe it is worth mentioning that they can keep APM working and only disable ASM by adding `DD_APPSEC_ENABLED=false` in their deployments. Something like this:
```suggestion
3. Restart the services you want to remove instrumentation for.

Note that you can disable ASM while keeping APM up by adding the `DD_APPSEC_ENABLED=false` environment variable to your deployments.

```

Ofc feel free to rework this

File: content/en/security/application_security/enabling/single_step/threat_detection.md [hunk 13]
```diff
@@ -0,0 +1,420 @@
...
+   ```shell
+   systemctl restart docker
+   ```
+   Or use the equivalent for your environment.
+
+{{% /tab %}}
+
+{{% tab "Kubernetes" %}}
+
+1. Under `apm:`, remove `instrumentation:` and all following configuration in `datadog-values.yaml`.
```
Comment (eliottness) [inline content/en/security/application_security/enabling/single_step/threat_detection.md:407]: This completes the suggestions I've put higher

```suggestion
1. Under `apm:`, remove `instrumentation:` and all following configuration in `datadog-values.yaml`.
2. Remove `asm:` and all underlying values in `datadog-values.yaml`.
```

**Comment by aliciascott** [comment]
Waiting on https://github.com/DataDog/helm-charts/pull/1337

**Pull request merged**